### PR TITLE
Upgrade to coap-lite 0.8

### LINKED ISSUE #70: Observe requests crash server if the client uses an empty vec![] to represent Observe: 0
The CoAP spec (RFC 7252) allows for options using the uint type to be encoded as an empty vector for 0 but this crashes here:

https://github.com/Covertness/coap-rs/blob/69e40da82ea90e60477cd66a466894e1448c9598/src/observer.rs#L85

coap-lite upstream has been upgraded to have proper handling for these types and we can fix the issue by simply upgrading to 0.8 (PR incoming)

### PR DESCRIPTION
**Pull request by jasta**
This new release contains a few minor API improvements, however
it also fixes #70 by upgrading to proper Option uint parsing support.

### CODE REVIEW

**Comment by jasta** [comment]
Hmm the build check is puzzling.  It's saying the new version of coap-lite can't compile but it works fine for me locally?  Looks like there's an issue in coap-lite that depends (maybe accidentally) on a PartialEq impl with slices that was added in 1.48.0 but the version used in 1.46.0.  Perhaps we can upgrade the continuous integration?

**Comment by jasta** [comment]
Actually yeah looks like maybe we should be using rustup here: 

https://github.com/japaric-archived/rust-everywhere/blob/ea191c8ba817a56d1911b08213a650ae67e568b5/appveyor.yml#L36-L43

File: src/observer.rs [hunk 1]
```diff
@@ -424,8 +419,8 @@ mod test {
     async fn request_handler(req: CoapRequest<SocketAddr>) -> Option<CoapResponse> {
         match req.get_method() {
             &coap_lite::RequestType::Get => {
-                let observe_option = req.message.get_observe().unwrap();
-                assert_eq!(observe_option[0], ObserveOption::Deregister as u8);
+                let observe_option = req.message.get_observe_value().unwrap();
```
Comment (Covertness) [inline src/observer.rs:422]: `observe_option` is wrong type. The line should be `let observe_option = req.get_observe_flag().unwrap().unwrap();`

**Comment by Covertness** [comment]
> Actually yeah looks like maybe we should be using rustup here:
> 
> https://github.com/japaric-archived/rust-everywhere/blob/ea191c8ba817a56d1911b08213a650ae67e568b5/appveyor.yml#L36-L43
> 
> and:
> 
> https://github.com/starkat99/appveyor-rust/blob/e14bc006cfdd643df27a4ec4553d96dc963ec390/appveyor.yml#L104-L109

You can have a try

**Comment by jasta** [comment]
Seems to be working great now

**Pull request merged**